com.unity.netcode.gameobjects/Runtime/Messaging/DeferredMessageManager.cs

```diff
@@ -113,14 +113,14 @@ public virtual void ProcessTriggers(IDeferredNetworkMessageManager.TriggerType t
                 // processed before the object is fully spawned. This must be the last thing done in the spawn process.
                 if (triggers.TryGetValue(key, out var triggerInfo))
                 {
+                    triggers.Remove(key);
```

Comment:
Why the move? (for context not that I think it shouldn't move :) )

Reply:
If the handler for a trigger tried to add a new trigger of the same type, the `Remove()` after executing the trigger would just delete it and the trigger would never actually get executed.